File: advent_ex/lib/day_one.ex
defmodule AdventEx.DayOne do
  def tasks(day_input) do
    IO.puts("Day One, part 1: #{part_one(day_input)}")
    IO.puts("Day One, part 2: #{part_two(day_input)}")
  end

  def part_one(day_input) do
    as_ints =
      day_input
      |> Enum.map(fn x -> x
      |> Integer.parse()
      |> elem(0) end)
      
    shifted = as_ints |> Enum.slice(Range.new(1, length(as_ints)))
    Enum.zip([as_ints, shifted]) |> Enum.filter(fn {x, y} -> x < y end) |> length()
  end

  def part_two(day_input) do
    as_ints =
      day_input
      |> Enum.map(fn x -> x |> Integer.parse() |> elem(0) end)
      |> Enum.chunk_every(3, 1)
      |> Enum.map(&Enum.sum/1)

    shifted = as_ints |> Enum.slice(Range.new(1, length(as_ints)))
    Enum.zip([as_ints, shifted]) |> Enum.filter(fn {x, y} -> x < y end) |> length()
  end
end

File: src/main/scala/advent2021/dayOne.scala
package advent2021

object dayOne {
    def partOne(input: Seq[String]) = {
        val as_ints = input.map(_.toInt)
        val shifted = as_ints.slice(1, as_ints.length)
        as_ints.zip(shifted).map { case (a, b) => a < b }.count(_ == true)
    }

    def partTwo(input: Seq[String]) = {
        val as_ints = input.map(_.toInt)
        val normalized = as_ints.sliding(3).map(_.sum).toList
        val shifted = normalized.slice(1, normalized.length)
        normalized.zip(shifted).map { case (a, b) => a < b }.count(_ == true)
    }
}